ocaml-lsp-server/src/ocamlformat_rpc.ml

```diff
@@ -69,26 +66,26 @@ end = struct
 
   let create ~logger ~bin () =
     let bin = Fpath.to_string bin in
-    let pid, stdout, stdin =
-      let stdin_i, stdin_o = Unix.pipe () in
-      let stdout_i, stdout_o = Unix.pipe () in
+    let* pid, stdout, stdin =
+      let* stdin_i, stdin_o = Lev_fiber.Io.pipe ~cloexec:true () in
+      let+ stdout_i, stdout_o = Lev_fiber.Io.pipe ~cloexec:true () in
       let pid =
-        Spawn.spawn ~prog:bin ~argv:[ bin ] ~stdin:stdin_i ~stdout:stdout_o ()
+        Spawn.spawn ~prog:bin ~argv:[ bin ]
+          ~stdin:(Lev_fiber.Fd.fd (Lev_fiber.Io.fd stdin_i))
+          ~stdout:(Lev_fiber.Fd.fd (Lev_fiber.Io.fd stdin_o))
```

Comment:
Shouldnt this be:
```suggestion
          ~stdout:(Lev_fiber.Fd.fd (Lev_fiber.Io.fd stdout_o))
```
From my understanding, and what it replaces, it seems so. However, when I make the change, the server crash with a `Sys_blocked io`...

Reply:
Thanks. This is fixed.

Reply:
How do you get the Sys_blocked io error?

Reply:
Unfortunately `make test` does not display well errors/logging, so I pinned my development version of `ocaml-lsp` in a switch, open vs-code on this switch, and hover a value or type with my mouse. Then, I can see the errors in the bottom panel, `Output` tab, selecting `OCaml Language Server` in the list.

![image](https://user-images.githubusercontent.com/34110029/152943822-72b5de32-7689-4f2d-be4c-50dd6d99e22c.png)

Reply:
Also, lines 159 and 161 of `ocamlformat_rpc.ml` should be "inverted" to un-`Disable` the rpc, to get the error.

I once committed that (and a new test for formatted output on hover) but it got erase by one of your force-push, I don't know if that was intentional.

Reply:
> Also, lines 159 and 161 of ocamlformat_rpc.ml should be "inverted" to un-Disable the rpc, to get the error.

Yes indeed. I forgot about that.

> I once committed that (and a new test for formatted output on hover) but it got erase by one of your force-push, I don't know if that was intentional.

Whoops that was completely accidental. Could you send a separate PR for the test though?

Reply:
Okay, I think I fixed the blocking issue.